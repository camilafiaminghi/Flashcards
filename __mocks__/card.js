const card = {
  question: 'What is React?',
  answer: 'A library for managing user interfaces'
}

export default card
