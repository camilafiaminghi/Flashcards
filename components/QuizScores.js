import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
import { textColorInverse, textColor, pColor, pColorLight, pColorDark, sColor, sColorLight, sColorDark } from '../utils/colors'
import { clearLocalNotification, setLocalNotification } from '../utils/helpers'

export default class QuizScores extends Component {
	static propTypes = {
		scored: PropTypes.number.isRequired,
		len: PropTypes.number.isRequired
	}

	componentDidMount() {
		clearLocalNotification()
			.then(setLocalNotification)
	}

	render(){
		const { scored, len } = this.props
		const percentageScored = Math.floor(scored * 100 / len)

		return (
			<View style={styles.container}>
				<View style={styles.row}>
					<Text style={[styles.detail, {fontSize: 20}]}>{scored}/{len}</Text>
				</View>

				<View style={[styles.row, {marginRight: -10}]}>
					<Text style={[styles.text, {fontSize: 90, fontWeight: 'bold', marginRight: -2}]}>{percentageScored}</Text>
					<Text style={[styles.text, {fontSize: 60}]}>%</Text>
				</View>

				<Text style={[styles.text, {fontSize: 30}]}>Scored</Text>
			</View>
		)
	}
}

/* STYLES */
export const styles = StyleSheet.create({
  container: {
    flex: 1,
  	flexGrow: 1,
  	alignItems: 'center',
  	justifyContent: 'center'
  },
  row: {
  	flexDirection: 'row',
  	alignItems: 'center',
  	marginRight: 10,
		marginLeft: 10
  },
  text: {
  	textAlign: 'center',
		color: textColor
  },
	detail: {
  	color: textColor
  }
})
