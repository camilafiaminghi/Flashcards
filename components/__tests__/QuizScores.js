import React, { Component } from 'react'
import { Text } from 'react-native'
import renderer from 'react-test-renderer'
import QuizScores from '../QuizScores'
import card from '../../__helpers__/card'

jest.useFakeTimers()

let wrapper
let instance
let component
const props = {
	scored: 5,
	len: 10
}

describe('<QuizScores />', () => {

	beforeEach(() => {
		wrapper = renderer.create(<QuizScores {...props} />)
		component = shallow(<QuizScores {...props} />)
	})

	it('matches snapshot', () => {
    expect(wrapper.toJSON()).toMatchSnapshot()
  });

	it('should render', () => {
		expect(wrapper).toBeTruthy()
	})

	it('should shows a 50% scored', () => {
		const text = component.find(Text).get(1)
		expect(text.props.children).toEqual(50)
	})
})
