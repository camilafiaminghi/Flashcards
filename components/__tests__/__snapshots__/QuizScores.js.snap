// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuizScores /> matches snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexGrow": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginLeft": 10,
        "marginRight": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#333333",
          },
          Object {
            "fontSize": 20,
          },
        ]
      }
    >
      5
      /
      10
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginLeft": 10,
          "marginRight": 10,
        },
        Object {
          "marginRight": -10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#333333",
            "textAlign": "center",
          },
          Object {
            "fontSize": 90,
            "fontWeight": "bold",
            "marginRight": -2,
          },
        ]
      }
    >
      50
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#333333",
            "textAlign": "center",
          },
          Object {
            "fontSize": 60,
          },
        ]
      }
    >
      %
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#333333",
          "textAlign": "center",
        },
        Object {
          "fontSize": 30,
        },
      ]
    }
  >
    Scored
  </Text>
</View>
`;
